pkg/bufferdecoder/eventsreader.go

```diff
@@ -309,52 +309,35 @@ func readStringFromBuff(ebpfMsgDecoder *EbpfDecoder) (string, error) {
 	return string(res), nil
 }
 
-// readSunPathFromBuff reads a null-terminated string from the eBPF buffer, up to `max` bytes.
-// Characters are read one by one until a NUL byte or the max limit is reached.
-// If the first byte is NUL and the second is not, it's treated as an abstract socket and
-// the first byte is replaced with '@'.
-// After reading, the decoder cursor advances past `max` bytes in the buffer.
-func readSunPathFromBuff(decoder *EbpfDecoder, max int) (string, error) {
-	if max <= 0 {
-		return "", errfmt.Errorf("max to decode sun_path must be greater than 0")
-	}
-
+// readVarStringFromBuffer reads a null-terminated string from the ebpf buffer where the size is not
+// known. The helper will read from the buffer char-by-char until it hits the null terminator
+// or the given max length. The cursor will then skip to the point in the buffer after the max length.
+func readVarStringFromBuffer(decoder *EbpfDecoder, max int) (string, error) {
 	var err error
 	var char int8
 	res := make([]byte, 0, max)
 
-	count := 0
-	for i := 0; i < max; i++ {
-		err = decoder.DecodeInt8(&char)
-		if err != nil {
-			return "", errfmt.Errorf("error reading sun_path at index %d out of %d: %v", i, max, err)
-		}
-		count++
-
-		if char == 0 {
-			// char as NUL may signal the end of the string or an abstract socket
-			// https://elixir.bootlin.com/linux/v6.13.4/source/net/unix/af_unix.c#L72
-			// https://man7.org/linux/man-pages/man7/unix.7.html
-			if i > 0 {
-				// NUL found after the first char means the end of the string
-				break
-			}
-		}
+	err = decoder.DecodeInt8(&char)
+	if err != nil {
+		return "", errfmt.Errorf("error reading null terminated string: %v", err)
+	}
 
+	count := 1 // first char is already decoded
+	for char != 0 && count < max {
 		res = append(res, byte(char))
-	}
 
-	if res[0] == 0 {
-		if len(res) == 1 {
-			res = []byte{} // empty string
-		} else {
-			// abstract socket - res[0] = NUL && res[1] != NUL
-			// https://elixir.bootlin.com/linux/v6.13.4/source/net/unix/af_unix.c#L3438
-			res[0] = '@'
+		// decode next char
+		err = decoder.DecodeInt8(&char)
+		if err != nil {
+			return "", errfmt.Errorf("error reading sun_path at index %d out of %d: %v", count, max, err)
 		}
+		count++
 	}
 
-	decoder.MoveCursor(max - count) // move cursor to the desired endpoint
+	// The exact reason for this Trim is not known, so remove it for now,
```

Comment:
We now know the exact reason of the commented out TrimLeft: to get rid of the NUL prefix that signify abstract sockets.

The kernel communicates it correctly when we dump `/proc/net/unix`, so should we when decoding. But as we already discussed, there are corner cases.

Reply:
not so sure because we still terminate at the first null in the reverted logic.